### ISSUE TITLE
move src/stan/interface to src/stan/interface_callbacks

### ISSUE THREAD

**Issue by syclik** [stan-dev/rstan#179]
Deal with change to Stan C++ API in 4e6fd1d8

**Comment by bgoodri** [comment]
Move when? I would like to try to keep rstan on GitHub in an installable state most of the time. If we moved things like this now, then someone would have to also install GitHub's StanHeaders after pointing it to develop, which does not work with devtools::install_github. So I would rather do this in a branch and / or wait until close to the next release. If it is just changing includes, then it shouldn't be a big deal.

**Comment by syclik** [comment]
Ben, I'm not sure what your email means, especially the last statement.

Isn't this why you'd want to keep StanHeaders at a tag and use that?

**Comment by bgoodri** [comment]
The v2.7.0 tag and master are pretty much the same thing. But if we have to
change StanHeaders to point at develop in order for rstan develop to
install, then someone would basically have to be a git user to install
rstan from GitHub.

**Comment by syclik** [comment]
no need to change StanHeaders until you're ready, right? Can't you just
leave the pull request for RStan open until you're ready for it?

**Comment by bgoodri** [comment]
Yes, that is what I was attempting to confirm.  Ideally, something like
this could wait until the week of the next release.

**Comment by syclik** [comment]
I'm still confused. Why not have Stan update when it updates? Just don't
update StanHeaders until you need to?

**Comment by bgoodri** [comment]
I was just saying that installing rstan via devtools::install_github will
use StanHeaders from CRAN so when we move paths from interface to
interface_callbacks, rstan won't easily build (ideally for a short amount
of time until the next StanHeaders is on CRAN).

**Comment by syclik** [comment]
Quick q: can you request that a certain version of rstan installed from
github requires a particular version of Stan headers?

**Comment by bgoodri** [comment]
Based on the version numbers in the DESCRIPTION files, yes.

But installing StanHeaders from GitHub is not easy because of the subrepos.
You pretty much have to clone it, grab the subrepos, build the R package
locally, and install it.
